tests/app/questionnaire/routing/test_operators.py

```diff
@@ -0,0 +1,291 @@
+from datetime import datetime, timedelta
+
+import pytest
+from dateutil.relativedelta import relativedelta
+
+from app.questionnaire.routing.helpers import datetime_as_midnight
+from app.questionnaire.routing.operator import Operator
+from app.questionnaire.rules import convert_to_datetime
+
+now = datetime.utcnow()
+now_as_yyyy_mm_dd = now.strftime("%Y-%m-%d")
+now_as_yyyy_mm = now.strftime("%Y-%m")
+now_as_yyyy = now.strftime("%Y")
+
+equals_operations = [
+    [(0.5, 0.5), True],
+    [(1.0, 1), True],
+    [(3, 3), True],
+    [(now, now), True],
+    [(0.5, 0.7), False],
+    [(1.0, 3), False],
+    [(3, 7), False],
+    [(now, datetime.utcnow()), False],
+    [("Yes", "Yes"), True],
+    [("CaseInsensitive", "caseInsensitive"), True],
+    [(None, None), True],
+    [(True, True), True],
+    [("Yes", "No"), False],
+    [(None, 1), False],
+    [(True, False), False],
+]
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    equals_operations,
+)
+def test_operation_equal(operands, expected_result):
+    operator = Operator(Operator.EQUAL)
+    assert operator.evaluate(operands) is expected_result
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    equals_operations,
+)
+def test_operation_not_equal(operands, expected_result):
+    operator = Operator(Operator.NOT_EQUAL)
+    assert operator.evaluate(operands) is not expected_result
+
+
+greater_than_and_less_than_operations = [
+    [(0.7, 0.5), True],
+    [(2, 1.0), True],
+    [(7, 3), True],
+    [(datetime.utcnow(), now), True],
+    [(0.5, 0.7), False],
+    [(1.0, 2), False],
+    [(3, 7), False],
+    [(now, datetime.utcnow()), False],
+]
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    greater_than_and_less_than_operations,
+)
+def test_operation_greater_than(operands, expected_result):
+    operator = Operator(Operator.GREATER_THAN)
+    assert operator.evaluate(operands) is expected_result
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    greater_than_and_less_than_operations,
+)
+def test_operation_less_than(operands, expected_result):
+    operator = Operator(Operator.LESS_THAN)
+    assert operator.evaluate(operands) is not expected_result
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    [
+        [(0.7, 0.5), False],
+        [(2, 1.0), False],
+        [(7, 3), False],
+        [(datetime.utcnow(), now), False],
+        [(0.5, 0.7), True],
+        [(1.0, 2), True],
+        [(3, 7), True],
+        [(now, datetime.utcnow()), True],
+        [(0.5, 0.5), True],
+        [(1.0, 1), True],
+        [(3, 3), True],
+        [(now, now), True],
+    ],
+)
+def test_operation_less_than_or_equal(operands, expected_result):
+    operator = Operator(Operator.LESS_THAN_OR_EQUAL)
+    assert operator.evaluate(operands) is expected_result
+
+
+@pytest.mark.parametrize(
+    "operands, expected_result",
+    [
+        [(0.7, 0.5), True],
```

Comment:
As above, I don't see order in this list

Reply:
https://github.com/ONSdigital/eq-questionnaire-runner/pull/629#discussion_r681055227